drivers/entropy/entropy_stm32.c

```diff
@@ -140,12 +140,16 @@ static int entropy_stm32_suspend(void)
 	LL_RNG_SetAesReset(rng, 1);
 #endif /* CONFIG_SOC_STM32WB09XX */
 
-#ifdef CONFIG_SOC_SERIES_STM32WBAX
-	uint32_t wait_cycles, rng_rate;
-
-	if (LL_PKA_IsEnabled(PKA)) {
+/* PKA module isn't currently supported in Zephyr, but it could be used outside Zephyr */
+#if defined(PKA)
+	if (__HAL_RCC_PKA_IS_CLK_ENABLED() && LL_PKA_IsEnabled(PKA)) {
+		/* PKA needs RNG clock, so exit here if in use */
 		return 0;
```

Comment:
Unlock semaphore for WBX and H7 dual core devices?
```suggestion
#if defined(CONFIG_SOC_SERIES_STM32WBX) || defined(CONFIG_STM32H7_DUAL_CORE)
		z_stm32_hsem_unlock(CFG_HW_RNG_SEMID);
#endif /* CONFIG_SOC_SERIES_STM32WBX || CONFIG_STM32H7_DUAL_CORE */
		return 0;
```
**(edited) actually only needed for WBXX.**